.gitignore

```diff
@@ -30,3 +30,4 @@ docs/VERSION
 docs/GITCOMMIT
 docs/changed-files
 autogen/
+Dockerfile
```

Comment:
@tianon can you elaborate on why we would want to do that? Seems like there are two choices:
1 - check-in something that's close to what we use and then explain to people why they shouldn't be modifying it and should be modifying the template file over, and over and over
2 - check-in something that is just a comment - like:
```
# Please use 'make ....' to generate the real version of this file
```
I guess 2 would be ok but I would hope our Makefile or README would give the same info.  Just feels like not having it would lead to less confusion.